core/internal/processlib/processlib_linux.go

```diff
@@ -0,0 +1,26 @@
+package processlib
+
+import (
+	"os"
+	"syscall"
+)
+
+const (
+	// Linux syscall for prctl, see x/sys/unix SYS_PRCTL
+	PRCTL_SYSCALL = 157
+	// Set the parent-death signal of the calling process, see uapi/linux/prctl.h
+	PR_SET_PDEATHSIG = 1
+)
+
+func ShutdownOnParentExit(parentPid int) bool {
+	_, _, errno := syscall.Syscall(
+		uintptr(PRCTL_SYSCALL), uintptr(PR_SET_PDEATHSIG), uintptr(syscall.SIGKILL), 0)
+	if errno != 0 {
+		os.Exit(127 + int(errno))
```

Comment:
could you add a comment about the math of this error code?

Reply:
just convention to pass detailed errno in top bits. but ill add a comment